test/basicTest.js

```diff
@@ -35,6 +35,13 @@ describe("basic tests", function() {
       "<iframe class='iframeTarget'/>",
       "</body>",
       "</html>"
+    ].join("\n"),
+    requiredJS = [
+      "var el = document.createElement('div');",
+      "el.id = \"test-shadow\";",
+      // "var shadow = el.attachShadow({ mode: 'open' })", // sadly shadow dom not working in jsdom
```

Comment:
I really want to test with shadow root at first, but jsdom not working with shadow dom. Maybe some day we can use smth like headless-chrome for testing

Reply:
[puppeteer](https://github.com/GoogleChrome/puppeteer)